addons/point_of_sale/report/pos_order_report.py

```diff
@@ -40,19 +40,30 @@ class PosOrderReport(models.Model):
 
     def _select(self):
         return """
+            -- The purpose of this CTE is to map each "pos_order_line" to the "payment_method_id" corresponding to its "pos_order"
+            -- considering we always show the first "payment_method_id"
+            WITH payment_method_by_order_line AS (
+                SELECT
+                    pol.id AS pos_order_line_id,
+                    (array_agg(pm.payment_method_id))[1] AS payment_method_id
+                FROM pos_order_line pol
+                LEFT JOIN pos_order po ON (po.id = pol.order_id)
+                LEFT JOIN pos_payment pm ON (pm.pos_order_id=po.id)
+                GROUP BY pol.id
+            )
             SELECT
-                MIN(l.id) AS id,
-                COUNT(*) AS nbr_lines,
+                l.id AS id,
+                1 AS nbr_lines, -- number of lines in order line is always 1
                 s.date_order AS date,
-                SUM(l.qty) AS product_qty,
-                SUM(l.qty * l.price_unit / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) AS price_sub_total,
-                SUM(ROUND((l.price_subtotal_incl) / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END, cu.decimal_places)) AS price_total,
-                SUM((l.qty * l.price_unit) * (l.discount / 100) / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END) AS total_discount,
+                l.qty AS product_qty,
+                l.qty * l.price_unit / CASE COALESCE(s.currency_rate, 0) WHEN 0 THEN 1.0 ELSE s.currency_rate END AS price_sub_total,
```

Comment:
```suggestion
                l.qty * l.price_unit / COALESCE(NULLIF(s.currency_rate, 0), 1.0) END AS price_sub_total,
```
Maybe we can take advantage of NULLIF and it will add more readability.

Reply:
applied, thank you.